addons/website_event/static/src/snippets/s_events/000.js

```diff
@@ -13,11 +13,10 @@ const DynamicSnippetEvents = DynamicSnippet.extend({
      */
     _getSearchDomain: function () {
         const searchDomain = this._super.apply(this, arguments);
-        const filterByTagIds = JSON.parse(this.$el.get(0).dataset.filterByTagIds || '[]');
-        if (filterByTagIds.length > 0) {
-            searchDomain.concat(Array(filterByTagIds.length-1).fill('&'));
-            for (const tag of filterByTagIds) {
-                searchDomain.push(['tag_ids', 'in', tag]);
+        const filterByTagIds = this.$el.get(0).dataset.filterByTagIds;
+        if (filterByTagIds) {
+            for (const tag of JSON.parse(filterByTagIds).map(eventTag => eventTag.id)) {
+                searchDomain.push(['tag_ids', 'in', [tag]]);
```

Comment:
Quick question on this.

Looks like here:
https://github.com/odoo/odoo/blob/master/addons/website_event/models/event_event.py#L498

We add some specific code to group the tags by categories to apply OR conditions in between tags of the same category.
Don't we need the same kind of behavior here?

I know it's not related to your changes, but it's worth checking as we are modifying this.